### PR TITLE
feat(popover): Add hint support

### PR DESCRIPTION
**Pull request by TatianaFomina**
It becomes possible to display a hint near popover item. 
Only works in desktop popover. 

image

This pull request is required for further progress on making Inline Toolbar render via Popover like Toolbar and Block Tunes

image

### CODE REVIEW

File: src/components/utils/popover/components/popover-item/popover-item-default/popover-item-default.ts [hunk 1]
```diff
@@ -86,11 +93,14 @@ export class PopoverItemDefault extends PopoverItem {
...
    * @param params - popover item construction params
+   * @param renderParams - popover item render params.
+   * The parameters that are not set by user via popover api but rather depend on technical implementation
    */
-  constructor(params: PopoverItemDefaultParams) {
+  constructor(params: PopoverItemDefaultParams, renderParams?: PopoverItemRenderParams) {
     super();
 
     this.params = params;
+    this.hintPosition = renderParams?.hint?.position || 'right';
```
Comment (neSpecc) [inline src/components/utils/popover/components/popover-item/popover-item-default/popover-item-default.ts]: seems redundant. Why not just use `renderParams.hint.position`?

File: src/components/utils/popover/components/popover-item/popover-item.types.ts [hunk 2]
```diff
@@ -152,3 +183,19 @@ export type PopoverItemParams =
   PopoverItemSeparatorParams |
   PopoverItemHtmlParams;
 
+
+/**
+ * Popover item render params.
```
Comment (neSpecc) [inline src/components/utils/popover/components/popover-item/popover-item.types.ts:168]:
```suggestion
```

File: src/components/utils/popover/components/popover-item/popover-item-default/popover-item-default.ts (hunk 1 above)
Comment (TatianaFomina) [reply src/components/utils/popover/components/popover-item/popover-item-default/popover-item-default.ts]: I do not want to let users set the position manually

Comment (TatianaFomina) [reply src/components/utils/popover/components/popover-item/popover-item-default/popover-item-default.ts]: so, seem like we need to split parameters that can be altered by users and the ones that are 'technical'

**Review by gohabereg — APPROVED** [review]

**Review by ilyamore88 — APPROVED** [review]

File: src/components/utils/popover/components/popover-item/popover-item.types.ts [hunk 3]
```diff
@@ -152,3 +183,25 @@ export type PopoverItemParams =
...
+   */
+  hint?: {
+    /**
+     * Hint position relative to the item
+     */
+    position?: HintPosition;
+
+    /**
+     * If false, hint will not be rendered.
+     * True by default
```
Comment (neSpecc) [inline src/components/utils/popover/components/popover-item/popover-item.types.ts]:
```suggestion
     * If false, hint will not be rendered.
     * True by default
     * Used to disable hints on mobile popover
```

Comment (TatianaFomina) [reply src/components/utils/popover/components/popover-item/popover-item.types.ts]: fixed

**Review by neSpecc — APPROVED** [review]

**Pull request merged**